### ISSUE TITLE
Media & Text: Linked images are missing a link text if there is no alt attribute

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#53622]
### Description

If the image in the Media & Text block is missing an alt attribute and the image is linked, screen readers will read out the link because there is no alternative link text.
For example, if the image is linked to a post or page on the site, it will use the link to that content.
If it is linked to the attachment page, a screen reader may read "link, http://63.local/wp-content/uploads/2023/08/Asset-1.png". If it is linked to the media file, it will read the name of the media file.

It would be difficult to come up with a default link text that would work for all scenarios.
Can we force the alt attribute to be added if the image is linked?

### Step-by-step reproduction instructions

1. Add a Media & text block
2. For other media, select an image without an alt attribute
3. Add a link in the block toolbar.
4. Save.
5. Go to the front of the website and confirm that the link has no link text, only the media file.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jorgefilipecosta** [comment]
Hi @carolinan,
I suppose we're facing the same issue with the image block too, right?

> Can we ensure that the alt attribute is added when the image is linked?

Normally, we don't adhere to a pattern of imposition on the user. I guess an option could be to temporarily link the image until the user adds an alt tag. If the user doesn't provide the alt tag, we could then remove the link. However, this seems a bit unusual.
What if we automatically attempted to generate a better alt attribute, something like "Image linking to the attachment page at: https://.." or "Image linking to the media file at: ...."? It's not perfect, but it would provide additional information at the very least.
